Type hint most export files

ksar-fd opened the pull request:
Version 2 of this. I'd borrowed a friend's laptop to make the previous PR, but had forgotten to set myself as author 😅 

Anyway I'll be adding to this bit by bit as I keep finding things that need annotating

ksar-fd committed:
Type hint most export files

ksar-fd committed:
Merge branch 'renpy:master' into type-hints

renpytom commented:
I'm looking through this, and I'm seeing a lot of reformatting files to styles I find ugly, like wrapping lines too short, and adding things like: 

https://github.com/renpy/renpy/pull/6216/files#diff-31a458f21cbb822ea0442ed6ce25a4c8236cd1db96a3b03b274681996e42387fR58

Which isn't something I want to see in Ren'Py. 

(I'm considering using black to reformat Ren'Py, but with much longer line lengths, and probably with the preview style.)

ksar-fd committed:
Annotate parser/lexer files and some of their dependencies

ksar-fd commented:
I'm using black to format the files. Should I configure it within this pr so it follows your style more closely?

ksar-fd commented:
> https://github.com/renpy/renpy/pull/6216/files#diff-31a458f21cbb822ea0442ed6ce25a4c8236cd1db96a3b03b274681996e42387fR58
> 

Also if you mean where it says
```py
_ = call()
```
I added that to shut up basedpyright's linter. I figured out a better way that doesn't disrupt the code, namely `reportUnusedCallResult `. I can fix it right away too

renpytom commented:
Yes on using black, though I need to figure out what I want the line length to be. I've been playing around with 120. 

I'm leaning on doing a massive reformat of the Python code towards the end of the current release cycle, after I've merged all of the pending PRs.  So this would wait for that to be merged. (Hopefully I'll be switching into release mode after I finish the current work on speeding up rendering.) 

I'll also need to audit these, so making it clear is going to be important.

ksar-fd committed:
Second round of annotating the exports module

ksar-fd committed:
Merge branch 'renpy:master' into type-hints

ksar-fd committed:
Third round of annotating the exports and their dependencies

ksar-fd committed:
Fourth round of annotating the exports and their dependencies

ksar-fd committed:
Merge branch 'master' into type-hints

ksar-fd commented:
Ok, I did as much as I could. I'm not terribly familiar with the codebase, so some of these are my best guess, but I tried to make it as coherent as possible

ksar-fd committed:
Merge branch 'master' into type-hints

Andykl committed:
Initial implementation of PEP 657

ksar-fd committed:
Merge branch 'master' into type-hints

Andykl committed:
Hotfix

Andykl committed:
Factor out exception handling code.

Andykl committed:
Chain exception handling errors instead of shadowing.

TDCMC committed:
Prevent shaping the text using fribidi when using harfbuzz

Also remove the unmapping of Arabic presentation forms, since it is now redundant with this change.

ksar-fd committed:
Merge branch 'master' into type-hints

ksar-fd commented:
Any updates abt this?

Andykl committed:
Abstract more stuff and move things around.

renpytom commented:
Scheduled to go in when I get some time to review it.

JamieDerg committed:
added Changes to enable console output in the launcher for macos

Andykl committed:
Add back `report_traceback` handling.

Andykl committed:
`stdout` not always connected to ANSI-compatiable console.

Andykl committed:
Add `sys.excepthook` with a new system, so crashes has better formatting.

Andykl committed:
Formatting fixes.

Andykl committed:
Do not print completly empty lines in frame formatting.

Andykl committed:
Fix non-ascii and CJK chars in source code.

Andykl committed:
Fix formatting in _exception screen.

Andykl committed:
Update console to work with new system.

Andykl committed:
Fixes for SyntaxError/ParseError.

Andykl committed:
Fix handling of non-ascii lines.

renpytom commented:
Question about this - did you write the types yourself, was this generated with a tool, or something else? 

Part of this is trying to figure out the right way to to deal with some of the formatting changes that are introduced in this PR. They make it a bit hard to review, and also I don't love some of the choices the formatter made. So I'm trying to figure out the right way to deal with that?

renpytom committed:
Merge pull request #6293 from JamieDerg/master

Enable console output in the Mac OS Launcher

renpytom committed:
Deal with cases where the ctc variable can't be set.

renpytom committed:
doc: Changelog.

Armanio committed:
Fix typos in russian translations

Andykl committed:
Merge pull request #6297 from Armanio/fix-ru-typos

Fix typos in russian translation

renpytom committed:
Merge pull request #6286 from TDCMC/master

Prevent shaping the text using fribidi when using harfbuzz

renpytom committed:
Merge pull request #6274 from renpy/pep-657

Initial implementation of PEP 657

renpytom committed:
Fix compiler warnings in renpybidi.

ksar-fd commented:
Yeah, I annotated the types myself, using my best guess based on how the function/class/variable is used.

I apologise for the formatting. I use  [black](https://github.com/psf/black) for automatic formatting, which I imagine isn't ideal since it's very opinionated and allows for little configuration. On the other hand, [ruff](https://docs.astral.sh/ruff/) allows for fine grained configuring, so it would be more ideal for what you're trying to do.

ksar-fd committed:
Merge branch 'master' into type-hints

renpytom commented:
Were you able to run this, especially on Python 3.12? I'm noticing a number of errors, ranging from undefined types being used to syntax errors, like: 

https://github.com/renpy/renpy/pull/6216/files#diff-220e3cc26db85b053ce4e42e2b8a45d825782497cb198d12a9f50aef3f5f48adR320

ksar-fd commented:
Yeah, I think it would be best if I split this up in a bunch of smaller PRs, so it's easier to manage on both ends

renpytom commented:
I'd suggest waiting on that. We're putting together some guidelines as to how Ren'Py typing should work, as we need a typing scheme that can last for a long time. That's going to be required for typing work to continue.

ksar-fd commented:
Alrighty

> https://github.com/renpy/renpy/pull/6216/files#diff-220e3cc26db85b053ce4e42e2b8a45d825782497cb198d12a9f50aef3f5f48adR320

And I'll test the changes and roll out fixes for typos like these
